src/anemoi/inference/inputs/ekd.py

```diff
@@ -128,298 +106,61 @@ def __init__(
         self._namer = namer if namer is not None else self.checkpoint.default_namer()
         assert callable(self._namer), type(self._namer)
 
-    def _filter_and_sort(self, data: Any, *, variables: List[str], dates: List[Any], title: str) -> Any:
-        """Filter and sort the data.
+    def create_state(self, *, date=None, variables=None, initial: bool = True) -> State:
```

Comment:
Why did you choose the default `initial` as True? Would it not make more sense the other way around?